### ISSUE TITLE
Maximized application does not change their state when we switch to another application in the same workspace

### ISSUE THREAD

**Issue by senntore** [pop-os/cosmic-comp#427]
When an application is maximized, if we switch to another application or restore a minimized application, the maximized application does not change it's state & hides the newly active application.
This is not a problem if the new active application is in another workspace but right now it happens int he same workspace as well.

**Comment by Drakulix** [comment]
Can you provide some more details? Is this with tiled or floating windows (both the maximized and restored/new windows)? Maximized windows are considered part of the floating layer, even if they previously were tiled.

As such this might be working as intended, if the newly created windows or restored windows end up on the tiling layer.

I can see however how this might be unintuitive, perhaps adding new windows to the tiling layer should always break the maximized state. Possibly @pop-os/ux has some thoughts on this?

**Comment by senntore** [comment]
Yes, this is an issue particularly for Auto Tilling mode. In floating mode, the active window always appear in front of the previously active window, so the maximized window does not affect it.

You can see the issue in this video below. Here I did three things after maximizing the browser in auto tilling mode.

First, I tried switching to the already open terminal window. 
And then I tried to restore the minimized terminal window. 
In both cases we cannot see the active window, probably because it's below the maximized window.

Lastly, I started Cosmic app store, it's new window did appear on the screen because the maximized window returned to it's previous state & tile position, which is what I expected in the previous two steps.

[2024-04-17_1080p_25fps_10pts.webm](https://github.com/pop-os/cosmic-comp/assets/58325574/c163a6bd-e269-4e3a-9a93-cbd34aaa8d66)

**Comment by Drakulix** [comment]
Not sure why this is only happening when spawning the store and not when launching or restoring the terminal window, so there is definitely an inconsistency here, that needs to be fixed. However I am not sure, which is the intended behaviour here, waiting for @pop-os/ux to confirm whether the window should be un-maximized or not.

Also The black border of the dock on unmaximize also looks wrong and glitched, so ping @wash2.

**Comment by maria-komarova** [comment]
The intended behavior is different for floating windows versus tiles windows. 
- If there is auto-tiling enabled the maximized window should restore into the layout. 
- If windows are floating the new window should show on top of the maximized window. 
Both work on my system when I check it. 
The only edge case I can think of right now is if someone uses Super+G with auto-tiling to float a window, then maximizes it and then launches a new window. The new window will tile. In this case it seems we should restore the maximized window as well to avoid loosing the window.

**Comment by maria-komarova** [comment]
The edge case is working as intended on my system as well.

**Comment by maria-komarova** [comment]
Switching focus seems to be weird though. Switching to another tiled window that is sitting behind the maximized window doesn't restore the maximized window on my system on the video above and on my system. It should restore and re-tile or float the maximized window.

I also found another edge case while testing things. If I have windows floating - maximize one, switch to another one, then use Super+Y to turn on auto-tiling, the maximized window remains on top when it should un-maximize and tile. It tiles if one purposefully restores it but it should just restore on switching the auto-tiling on (even if it was focused when auto-tiling was turned on).
